library/Network/CloudSeeder/Types.hs

```diff
@@ -70,7 +83,7 @@ instance FromJSON ParameterSpecs where
         defVal <- defParser
               -- try parsing as a double if parsing fails as a string
               <|> fmap (fmap (T.pack . show)) (defParser @Double)
-        return $ maybe (Required k) (Optional k) defVal
+        return $ maybe (Required k) (\v -> Optional k $ Just v) defVal
```

Comment:
This is very stylistic, and I’ve avoided this sort of comment before because I think it’s extremely subjective, but I would generally avoid using the `$` in the lambda here. I would just write `(\v -> Optional k (Just v))` (or `Optional k . Just`, but let’s stay simple here).

Why? Well, in my mind, there is a meaningful difference in intent between these two expressions:

```hs
foo x (bar y)
foo x $ bar y
```

The former is *very* explicit about applying `foo` to two arguments. The latter is doing the same thing, of course, but it reads a little differently: it creates a *function pipeline* of two functions, `foo x` and `bar`, and applies that pipeline to `y`. Think of it like writing `(foo x . bar) y`.

Often, this is what you want. The use of `$` after `return` in this expression is quite a bit more readable, since it directs the eye to focus more on the expression to the right of the dollar sign. It usually indicates that the expression to the right of the `$` is more important—it’s where the bulk of the logic is located—and the left of the `$` just contains some simple wrapper functions or secondary operations.

In this case, however, `Optional` is a data constructor, constructing a product type with two fields. In this case, my mental model of `Optional` fundamental involves “two things”; it’s basically a tuple. Using parentheses around `(Just v)` deemphasizes the importance of `Just v` (which, in this case, is hardly important at all) and emphasizes that we are constructing a two-valued datatype.

This might sound nitpicky—and perhaps it is, please tell me if you feel that way—but I think one of the reasons I and other Haskellers react negatively to people who want to “ban `$`” in favor of parens is because `$` is *meaningfully* different from parens. It expresses a certain sort of *programmer intent*. If the two things really were identical, `$` would be a little pointless, but they’re idiomatically used differently, and I think it’s helpful to respect those idioms.